src/main/java/seedu/address/model/person/ContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,94 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class ContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> nameKeywords;
+    private final List<String> telegramHandleKeywords;
+    private final List<String> emailKeywords;
+    private final List<String> studentStatusKeywords;
+    private final List<String> roleKeywords;
+    private final List<String> nicknameKeywords;
+
+    /**
+     * Constructs a {@code ContainsKeywordsPredicate}.
+     *
+     * @param nameKeywords keywords to match with name
+     * @param telegramHandleKeywords keywords to match with telegram handle
+     * @param emailKeywords keywords to match with email
+     * @param studentStatusKeywords keywords to match with student status
+     * @param roleKeywords keywords to match with role
+     * @param nicknameKeywords keywords to match with nickname.
+     */
+    public ContainsKeywordsPredicate(List<String> nameKeywords, List<String> telegramHandleKeywords,
+                                     List<String> emailKeywords, List<String> studentStatusKeywords,
+                                     List<String> roleKeywords, List<String> nicknameKeywords) {
+        this.nameKeywords = nameKeywords;
+        this.telegramHandleKeywords = telegramHandleKeywords;
+        this.emailKeywords = emailKeywords;
+        this.studentStatusKeywords = studentStatusKeywords;
+        this.roleKeywords = roleKeywords;
+        this.nicknameKeywords = nicknameKeywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        boolean containsNameKeywords = nameKeywords.stream()
+                .anyMatch(keyword -> StringUtil.containsIgnoreCase(person.getName().fullName, keyword));
+        boolean containsTelegramHandleKeywords = telegramHandleKeywords.stream()
+                .anyMatch(keyword -> StringUtil.containsIgnoreCase(person.getTelegramHandle().value, keyword));
```

Comment:
Name.equals wouldn't work here because it does not support substring searching

I thought about ways to resolve this but:

It would not make sense to modify all the field classes eg. Name class to include the containsIgnoreCase functionality since it's only used for find

Adding a getNameString method for person kinda de-abstracts name from person